diff --git a/config/base/kf5-qt5 b/config/base/kf5-qt5
index 2260aa6..c82e3fc 100644
--- a/config/base/kf5-qt5
+++ b/config/base/kf5-qt5
@@ -20,7 +20,7 @@ general/vc: -qt5
 general/shared-desktop-ontologies: -qt5
 
 # KDE Frameworks
-frameworks/kauth: kdesupport/polkit-qt-1
+#frameworks/kauth: kdesupport/polkit-qt-1
 kde/*: general/kdesupport-svn
 kde/workspace/plasma-workspace: general/libdbusmenu-qt
 
diff --git a/config/build/darwin-mavericks.cfg b/config/build/darwin-mavericks.cfg
index cb62c49..64f14f6 100644
--- a/config/build/darwin-mavericks.cfg
+++ b/config/build/darwin-mavericks.cfg
@@ -1,7 +1,9 @@
 [DEFAULT]
+scriptsLocation=/Users/marko/scripts
 opSys=darwin
 architecture=mavericks
 compiler=clang
+configurePlatformArgs=-DCMAKE_INSTALL_BUNDLEDIR="Applications/KF5" -DDATA_INSTALL_DIR="Library/Application Support" -DCMAKE_DISABLE_FIND_PACKAGE_X11=ON
 
 [General]
-installPrefix=/opt/kde/install/%(opSys)s/%(architecture)s/%(compiler)s/
\ No newline at end of file
+installPrefix=/opt/kde/install/%(opSys)s/%(architecture)s/%(compiler)s/
diff --git a/config/build/global.cfg b/config/build/global.cfg
index 4052215..9dbe1e1 100644
--- a/config/build/global.cfg
+++ b/config/build/global.cfg
@@ -55,7 +55,7 @@ updateMimeDatabaseCommand=%(updateMimeDatabaseExecutable)s share/mime
 deployInstallation=True
 
 [Test]
-testsEnabled=True
+testsEnabled=False
 ctestCountTestsCommand=%(ctestExecutable)s -N
 xvfbDisplayName=:99
 xvfbCommand=%(xvfbExecutable)s %(xvfbDisplayName)s -ac -screen 0 1600x1200x24+32
diff --git a/tools/environment-generator.py b/tools/environment-generator.py
index 37dcbae..e46a13c 100644
--- a/tools/environment-generator.py
+++ b/tools/environment-generator.py
@@ -31,7 +31,7 @@ environment = manager.generate_environment(True)
 neededVariables = [
 	'CMAKE_PREFIX_PATH', 'KDEDIRS', 'PATH', 'LD_LIBRARY_PATH', 'PKG_CONFIG_PATH', 'PYTHONPATH',
 	'PERL5LIB', 'QT_PLUGIN_PATH', 'QML_IMPORT_PATH', 'QML2_IMPORT_PATH', 'XDG_DATA_DIRS',
-	'XDG_CONFIG_DIRS', 'QMAKEFEATURES', 'XDG_CURRENT_DESKTOP'
+	'XDG_CONFIG_DIRS', 'QMAKEFEATURES', 'XDG_CURRENT_DESKTOP', 'DYLD_LIBRARY_PATH'
 ]
 
 # Generate the shell format environment file, suitable for sourcing
