diff --git interfaces/foregroundlock.cpp interfaces/foregroundlock.cpp
index 4f76c6a..edd8b3e 100644
--- interfaces/foregroundlock.cpp
+++ interfaces/foregroundlock.cpp
@@ -101,7 +101,7 @@ void lockForegroundMutexInternal() {
         // We already have the mutex
         ++recursion;
     }else{
-        internalMutex.lockInline();
+        internalMutex.lock();
         Q_ASSERT(recursion == 0 && holderThread == 0);
         holderThread = QThread::currentThread();
         recursion = 1;
@@ -134,7 +134,7 @@ void unlockForegroundMutexInternal() {
     if(recursion == 0)
     {
         holderThread = 0;
-        internalMutex.unlockInline();
+        internalMutex.unlock();
     }
 }
 }
